### PR TITLE
ci: Avoid CI failures from temp env file reuse

### PR DESCRIPTION
**Pull request by maflcko**
* Currently, running separate CI tasks at the same time may intermittently fail, because they race to read/write `/tmp/env`. Fix this by adding `$CONTAINER_NAME` to the file name.

* Also, add `$USER`, while touching the line, to allow different users to run the same CI task at the same time.

* Also, skip the git install if there is no need.

Ref: https://github.com/bitcoin/bitcoin/pull/29274

### CODE REVIEW

**Review by Sjors — APPROVED** [review]
ACK fa91bf2559d2e839592bf1dc1d423d5fb1c3573e

**Review by BrandonOdiwuor — APPROVED** [review]
ACK fa91bf2559d2e839592bf1dc1d423d5fb1c3573e

Looks good to me

File: .cirrus.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ base_template: &BASE_TEMPLATE
   << : *FILTER_TEMPLATE
   merge_base_script:
     # Unconditionally install git (used in fingerprint_script).
```
Comment (hebasto) [inline .cirrus.yml:57]: This comment got outdated, no?

**Review by hebasto — APPROVED** [review]
ACK fa91bf2559d2e839592bf1dc1d423d5fb1c3573e.

**Comment by m3dwards** [comment]
Post merge ACK https://github.com/bitcoin/bitcoin/commit/fa91bf2559d2e839592bf1dc1d423d5fb1c3573e.

I ran multiple CI jobs at the same time and observed the nicer naming in the /tmp directory.

**Pull request merged**